Source/JavaScriptCore/b3/air/AirLowerMacros.cpp

```diff
@@ -53,25 +53,63 @@ void lowerMacros(Code& code)
                 Kind oldKind = inst.kind;
 
                 Vector<Arg> destinations = computeCCallingConvention(code, value);
+                Vector<Arg> finalDestinations;
 
                 unsigned resultCount = cCallResultCount(code, value);
                 
                 Vector<ShufflePair, 16> shufflePairs;
                 bool hasRegisterSource = false;
-                unsigned offset = 1;
-                auto addNextPair = [&](Width width) {
+                unsigned airArgsOffset = 1; // Skip the Special Arg in CCall.
+                unsigned destinationsOffset = 0;
+                auto addNextPair = [&](Type type) {
+                    Width width = cCallArgumentRegisterWidth(type);
+                    // CCall takes (special, callee, *<results>, *<args>).
+                    // Here, we're iterating over the arguments (including the
+                    // callee), so skip over the results (if any).
+                    if (airArgsOffset == 2)
+                        airArgsOffset += resultCount;
+#if USE(JSVALUE32_64)
+                    auto splitHiLo = [] (Arg& arg, Arg& hi, Arg& lo) {
+                        if (arg.isTmpPair()) {
+                            hi = arg.tmpHi();
+                            lo = arg.tmpLo();
+                        } else if (arg.isCallArg()) {
+                            hi = arg;
+                            lo = Air::Arg::callArg(arg.offset() + bytesForWidth(Width64));
```

Comment:
not Width32?

Reply:
Thanks. Not only that, it's hi that should be offset by 4 bytes.